### ISSUE TITLE
Unable to log in to private registry after upgrading to Docker Desktop 3.3.1

### ISSUE THREAD

**Issue by sebastien-ma** [docker/for-mac#5617]
- [x ] I have tried with the latest version of Docker Desktop
  - [ x] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior
Set domain for a private registry in Preferences -> Resources -> Proxies -> Bypass proxy settings for these hosts & domains
Be able to log in to the private registry and pull images from it

### Actual behavior
Log in to the private registry failed with message "x509: certificate signed by unknown authority"

### Information

  - Is it reproducible? Yes.
  - Is the problem new? Yes, only apear after updating to 3.3.1.
  - Did the problem appear with an update? appears after updating to 3.3.1. Works fine in 3.2.2
  - macOS Version: Big Sur
  - Intel chip or Apple chip: Intel chip
  - Docker Desktop Version: 3.3.1

### Steps to reproduce the behavior

  1. Enable "Manual Proxy Configuration" and set private registry domain in no proxy value
  2. docker login to Docker Hub -> successful, docker login <private registry> -> failed
  2. Disable "Manual Proxy Configuration" 
  3. docker login to Docker Hub -> failed, docker login <private registry> -> successful

**Comment by stephen-turner** [comment]
Thanks for the report. Could you try the developer build posted by my colleague @djs55 at https://github.com/docker/for-mac/issues/5572#issuecomment-822302592 and report whether it fixes it? If not, please upload diagnostics and put the diagnostics id here. Thank you.

**Comment by ak-at-gvl** [comment]
I can confirm the problems described by @sebastien-ma !
Communication with a custom repository in LAN (sonatype nexus) failed with strange errors (500 Server error using HTTP, 502 Bad Gateway using HTTPS). Also trying to contact the custom repo from inside a container (e. g. `yarn install` using same sonatype nexus) failes with similar errors.

After downgrading to 3.2.2 everything worked as expected.

**Comment by pguso** [comment]
Downgrading to 3.2.2 worked also for me. I had to cleanup some folders on my mac

```
sudo rm -Rf /Applications/Docker.app
sudo rm -f /usr/local/bin/docker
sudo rm -f /usr/local/bin/docker-machine
sudo rm -f /usr/local/bin/docker-compose
sudo rm -f /usr/local/bin/docker-credential-desktop
sudo rm -f /usr/local/bin/docker-credential-ecr-login
sudo rm -f /usr/local/bin/docker-credential-osxkeychain
sudo rm -Rf ~/.docker
sudo rm -Rf ~/Library/Containers/com.docker.docker
sudo rm -Rf ~/Library/Application\ Support/Docker\ Desktop
sudo rm -Rf ~/Library/Group\ Containers/group.com.docker
sudo rm -f ~/Library/HTTPStorages/com.docker.docker.binarycookies
sudo rm -f /Library/PrivilegedHelperTools/com.docker.vmnetd
sudo rm -f /Library/LaunchDaemons/com.docker.vmnetd.plist
sudo rm -Rf ~/Library/Logs/Docker\ Desktop
sudo rm -Rf /usr/local/lib/docker
sudo rm -f ~/Library/Preferences/com.docker.docker.plist
sudo rm -Rf ~/Library/Saved\ Application\ State/com.electron.docker-frontend.savedState
sudo rm -f ~/Library/Preferences/com.electron.docker-frontend.plist
```

**Comment by sebastien-ma** [comment]
> Thanks for the report. Could you try the developer build posted by my colleague @djs55 at [#5572 (comment)](https://github.com/docker/for-mac/issues/5572#issuecomment-822302592) and report whether it fixes it? If not, please upload diagnostics and put the diagnostics id here. Thank you.

I tried the dev build 63310, doesn't seem to help. Still the same symptom. Not sure if I'm allowed to upload diagnostics. I'll do that if I can.

**Comment by zgramana** [comment]
Upgrading to 3.3.2 resolved this problem for me

**Comment by djs55** [comment]
@zgramana thanks for letting us know the problem is fixed in 3.3.2! It should be fixed for everyone, so I'll close this ticket. If it's still broken please open a new ticket. Thanks for your patience!
